sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -43,7 +43,15 @@ class DataFrameStatSuite extends FunSuite  {
     val singleColResults = df.stat.freqItems(Array("negDoubles"), 0.1)
     val items2 = singleColResults.collect().head
     items2.getSeq[Double](0) should contain (-1.0)
+  }
 
+  test("pearson correlation") {
+    val df = sqlCtx.sparkContext.parallelize(
+      Array.tabulate(10)(i => (i, 2 * i, i * -1.0))).toDF("a", "b", "c")
```

Comment:
fyi we have implicits to add toDF on Seq[Tuples], so you can just replace Array with Seq, and then remove all the sparkContext.parallelize stuff. Maybe do it for the frequent items above also.